File: src/Systems/C++/singleton.cpp
#include <iostream>

class ProcessRegistry {
public:
    static ProcessRegistry& instance() {
        static ProcessRegistry value;
        return value;
    }

    void increment() { ++count_; }
    [[nodiscard]] int count() const { return count_; }

    ProcessRegistry(const ProcessRegistry&) = delete;
    ProcessRegistry& operator=(const ProcessRegistry&) = delete;

private:
    ProcessRegistry() = default;
    int count_ = 0;
};

int main() {
    auto& first = ProcessRegistry::instance();
    auto& second = ProcessRegistry::instance();
    first.increment();
    std::cout << "same=" << std::boolalpha << (&first == &second) << '\n';
    std::cout << "count=" << second.count() << '\n';
}

File: src/Systems/Swift/singleton.swift
final class ProcessRegistry {
    static let shared = ProcessRegistry()
    private(set) var count = 0

    private init() {}

    func increment() {
        count += 1
    }
}

let first = ProcessRegistry.shared
let second = ProcessRegistry.shared
first.increment()
print("same=\(first === second)")
print("count=\(second.count)")
